### PR TITLE
Remove sync over async in response content stream

### LINKED ISSUE #11626: Blob upload fails when run in Blazor WebAssembly
**Describe the bug**
The `UploadAsync` call fails when run in Blazor WebAssembly. Here is my sample code:

```csharp
public async Task UploadPayload(Stream payload)
{
  BlobClient client = new BlobClient("myConnectionString", "container", "bla");
  await client.UploadAsync(payload);
}
```

**Expected behavior**
The call succeeds

**Actual behavior (include Exception or Stack Trace)**
I get the following error:
`System.Threading.SynchronizationLockException: Cannot wait on monitors on this runtime.`

**To Reproduce**
- Using latest VS 2019 preview (16.6 Preview 4)
- Installed 3.1.3 SDK

Create a new Blazor WebAssembly application (3.2. Preview5 version):
- dotnet new -i Microsoft.AspNetCore.Components.WebAssembly.Templates::3.2.0-preview5.20216.8
- dotnet new blazorwasm
- Add reference to `<PackageReference Include="Azure.Storage.Blobs" Version="12.4.1" />` package
- Update the `Pages/Counter.razor` as follows (Note: replace `myConnectionString` with a valid Blob storage connection string)
- Run the app, go to the `/counter` page and click the `Click Me` button

```razor
@page "/counter"
@using Azure.Storage.Blobs
@using System.IO

<h1>Counter</h1>

<p>Current count: @currentCount</p>

<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>

@code {
    private int currentCount = 0;

    private async Task IncrementCount()
    {
        using MemoryStream payload = new MemoryStream();
        using StreamWriter sr = new StreamWriter(payload);
        sr.WriteLine("Sample payload to upload");
        sr.Flush();
        payload.Position = 0;

// The above code is simply to simulate a payload
        BlobClient client = new BlobClient("myConnectionString", "container", "bla");
        await client.UploadAsync(payload);
    }
}
```

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs v.12.4.1
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): .NET Core Runtime: 3.1.3
OS: Win 10 Pro x64

 - IDE and version : VS 16.6 Preview4

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/11626

### CODE REVIEW

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs [hunk 1]
```diff
@@ -414,91 +393,5 @@ public override void Dispose()
...
-                }
-            }
-
-            private Stream Stream
-            {
-                get
-                {
-                    if (_contentStream == null)
-                    {
-                        return EnsureStreamAsync().GetAwaiter().GetResult();
```
Comment (pakrym) [inline sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:486]: This was being hit when running on Blazor

File: sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs [hunk 2]
```diff
@@ -414,91 +396,5 @@ public override void Dispose()
 
             public override string ToString() => _responseMessage.ToString();
         }
-
-        private class ContentStream : ReadOnlyStream
```
Comment (AlexanderSher) [inline sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:418]: What was the purpose of this type in the first place?

Comment (pakrym) [reply sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs:418]: Make content stream access lazy, idea was that we can postpone calling ReadAsStreamAsync until a read on the stream is called. Unfortunately in most cases a sync stream access would be performed first.

**Comment by heaths** [comment]
Would this also resolve #8233? Or was that only a problem with the mock transport streams?

**Comment by pakrym** [comment]
> Would this also resolve #8233? Or was that only a problem with the mock transport streams?

No, this PR doesn't fix that, that part is more about mock transport.

**Review by heaths — APPROVED** [review]

**Pull request merged**